src/survey/surveyObj.ts

```diff
@@ -0,0 +1,295 @@
+import { AuthGroup } from '../auth'
+import { SystemError } from '../error'
+import {
+  NodeDefBooleanProps,
+  NodeDefCodeProps,
+  NodeDefDecimalProps,
+  NodeDefFileProps,
+  NodeDefMeta,
+  NodeDefProps,
+  NodeDefPropsAdvanced,
+  NodeDefTaxonProps,
+  NodeDefTextProps,
+  NodeDefType,
+} from '../nodeDef'
+import { Survey, SurveyProps } from './survey'
+
+export abstract class ArenaObj<P> {
+  uuid: string
+  id?: number
+  props?: P
+
+  constructor(uuid: string) {
+    this.uuid = uuid
+  }
+}
+
+export abstract class NodeDefObj<P extends NodeDefProps = NodeDefProps> extends ArenaObj<P> {
```

Comment:
puff, this code smell to the 2000s, Java is back...

Reply:
yes, but writing nodeDef.parent is much simpler than writing Surveys.getNodeDefParent({ survey, nodeDef }) or Surveys.getNodeDefParent(nodeDef)(survey), isn't it?!